### PR TITLE
Add 4.0.2 release note

### PR DESCRIPTION
**Pull request by uglyengineer**
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: releases/release-4.0.2.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
++ TiDB
+    - 降低当集群中 CM-Skech 占用过多内存时，golang 内存分配带来的查询延迟。[#17545](https://github.com/pingcap/tidb/pull/17545)
+    - 缩短 TiKV 故障恢复时集群 QPS 的恢复时间。[#17681](https://github.com/pingcap/tidb/pull/17681)
+    - 为 partition 表上的查询支持聚合函数下推到 TiKV 或者 TiFlash coprocessor。[#17655](https://github.com/pingcap/tidb/pull/17655)
+    - 提升索引上等值条件的行数估算准确度。[#17611](https://github.com/pingcap/tidb/pull/17611)
+
+## Bug 修复
+
++ TiDB
+    - Fix the incorrect execution plan in plan cache after `tidb_isolation_read_engines` is changed. [#17570](https://github.com/pingcap/tidb/pull/17570)
```
Comment (TomShawn) [inline releases/release-4.0.2.md:72]: @uglyengineer 这里应该提中文吧？

File: releases/release-4.0.2.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    - 修复对 `STR_TO_DATE` 函数的 `%h` 解析和 MySQL 不兼容问题。[#17498](https://github.com/pingcap/tidb/pull/17498)
+    - 修复`tidb_replica_read` 设置成 `follower`，并且 region 的 leader 和 follower/learner 之间出现网络分区后，TiDB 发送的 request 一直重试的问题。[#17443](https://github.com/pingcap/tidb/pull/17443)
+    - 修复某些情况下 TiDB 一直 ping PD 的 follower 的问题。[#17947](https://github.com/pingcap/tidb/pull/17947)
+    - 修复老版本的 range partition 表无法在 4.0 集群中加载的问题。[#17983](https://github.com/pingcap/tidb/pull/17983)
+    - 修复当多个 region 的请求同时超时时整个 SQL 语句超时的问题。[#17585](https://github.com/pingcap/tidb/pull/17585)
+    - 修复解析日期类型的分隔符时和 MySQL 不兼容的问题。[#17501](https://github.com/pingcap/tidb/pull/17501)
+    - 修复少数情况下发给 TiKV 的请求错发给 TiFlash 的问题。[#18105](https://github.com/pingcap/tidb/pull/18105)
+
++ TiKV
+
```
Comment (Connor1996) [inline releases/release-4.0.2.md:111]:
```suggestion
    - 修改 status server 的内存安全问题。[#8101](https://github.com/tikv/tikv/pull/8101)
    - 修复 json 数字比较的精度丢失问题。[#8087](https://github.com/tikv/tikv/pull/8087)
    - 修改错误的慢查询日志。[#8050](https://github.com/tikv/tikv/pull/8050)
    - 修复 merge 可能导致 peer 无法被移除的问题。[#8048](https://github.com/tikv/tikv/pull/8048)
    - 修复 `tikv-ctl recover-mvcc`  未清除无效的悲观锁。[#8047](https://github.com/tikv/tikv/pull/8047)
    - 修复一些遗漏的 Titan 监控。[#7997](https://github.com/tikv/tikv/pull/7997)
    - 修复向 TiCDC 返回 `duplicated error` 的问题。[#7887](https://github.com/tikv/tikv/pull/7887)
```

File: releases/release-4.0.2.md (hunk 1 above)
Comment (uglyengineer) [reply releases/release-4.0.2.md]: 已经改正

**Comment by breeswish** [comment]
TODO: Need to add a notice (and the reference link) about the telemetry behaviour. Need to wait for https://github.com/pingcap/docs-cn/pull/3812

File: releases/release-4.0.2.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+TiDB 版本：4.0.2
+
+## 兼容性
+
++ TiDB
+
+    - 移除慢查询日志和 statement summary 表中的敏感信息 [#18130](https://github.com/pingcap/tidb/pull/18130)
+    - 禁止在 sequence 缓存中出现负数 [#18103](https://github.com/pingcap/tidb/pull/18103)
+    - `CLUSTER_INFO` 表中不再显示 tombstone 状态的 TiKV 和 TiFlash 结点 [#17953](https://github.com/pingcap/tidb/pull/17953)
```
Comment (TomShawn) [inline releases/release-4.0.2.md:18]:
```suggestion
    - `CLUSTER_INFO` 表中不再显示 tombstone 状态的 TiKV 和 TiFlash 节点 [#17953](https://github.com/pingcap/tidb/pull/17953)
```

File: releases/release-4.0.2.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+ 
+## Bug 修复
+
++ TiDB
+
+    - 修复 `tidb_isolation_read_engines` 更改后从 plan cache 中获取的执行计划不正确的问题 [#17570](https://github.com/pingcap/tidb/pull/17570)
+    - 修复某些情况下 `EXPLAIN FOR CONNECTION` 返回运行时错误的问题 [#18124](https://github.com/pingcap/tidb/pull/18124)
+    - 修复某些情况下 `last_plan_from_cache` 结果不正确的问题 [#18111](https://github.com/pingcap/tidb/pull/18111)
+    - 修复执行 plan cache 中的 `UNIX_TIMESTAMP()` 时的运行时错误 [#18002](https://github.com/pingcap/tidb/pull/18002) [#17673](https://github.com/pingcap/tidb/pull/17673)
+    - 修复 `HashJoin` 算子的孩子结点返回 `NULL` 类型的结果时，计算过程中的运行时错误 [#17937](https://github.com/pingcap/tidb/pull/17937)
```
Comment (TomShawn) [inline releases/release-4.0.2.md:86]:
```suggestion
    - 修复 `HashJoin` 算子的孩子节点返回 `NULL` 类型的结果时，计算过程中的运行时错误 [#17937](https://github.com/pingcap/tidb/pull/17937)
```

File: releases/release-4.0.2.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    - 对 leader 执行 `remove-peer` 操作时，让这个 operator 不等待超时，立刻失败 [#2551](https://github.com/pingcap/pd/pull/2551)
+    - 对 TiFlash 节点设置更合理的 store limit 配置默认值 [#2559](https://github.com/pingcap/pd/pull/2559)
+
++ TiFlash
+
+    - Coprocessor 支持新的聚合函数 `APPROX_COUNT_DISTINCT`
+    - 存储引擎中的粗糙索引默认开启
+    - 支持运行在 ARM 架构
+    - Coprocessor 支持 `JSON_LENGTH` 函数下推
+
```
[the remark this answers is not in the document]
Comment (amyangfei) [reply releases/release-4.0.2.md:60]: 需要有

File: releases/release-4.0.2.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    - 修复TiDB 和对象存储服务之间 `BACKUP`/`RESTORE` 失败的问题 [#17844](https://github.com/pingcap/tidb/pull/17844)
+    - 修复权限检查失败时的错误信息 [#17724](https://github.com/pingcap/tidb/pull/17724)
+    - 修复 `DELETE`/`UPDATE` 语句的 feedback 被错误的收集的问题 [#17843](https://github.com/pingcap/tidb/pull/17843)
+    - 禁止更改非 `AUTO_RANDOM` 表的 `AUTO_RANDOM_BASE` 值 [#17828](https://github.com/pingcap/tidb/pull/17828)
+    - 修复系统变量 `tidb_isolation_read_engines` 的值中没有 `tidb` 时某些系统表无法访问的问题 [#17719](https://github.com/pingcap/tidb/pull/17719)
+    - 修复 JSON 中大整数和浮点数比较的精度问题 [#17717](https://github.com/pingcap/tidb/pull/17717)
+    - 修复 `COUNT()` 函数的返回类型中 `DECIMAL` 不正确的问题 [#17704](https://github.com/pingcap/tidb/pull/17704)
+    - 修复 `HEX()` 函数的输入类型是二进制字符串时结果不正确的问题 [#17620](https://github.com/pingcap/tidb/pull/17620)
+    - 修复查询 `INFORMATION_SCHEMA.INSPECTION_SUMMARY` 表没有指定过滤条件时返回结果为空的问题 [#17697](https://github.com/pingcap/tidb/pull/17697)
+    - 修复 `ALTER USER` 语句使用哈希后的密码更新密码后，密码不符合预期的问题 [#17646](https://github.com/pingcap/tidb/pull/17646)
```
[the remark this answers is not in the document]
Comment (AilinKid) [reply releases/release-4.0.2.md:104]: 多谢， how about: 修复 `ALTER USER` 语句使用哈希后的密码更新用户信息后，密码不符合预期的问题

File: releases/release-4.0.2.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (amyangfei) [reply releases/release-4.0.2.md:60]: added

File: releases/release-4.0.2.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (TomShawn) [reply releases/release-4.0.2.md:117]: Good

File: releases/release-4.0.2.md [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
++ TiFlash
+
+    - 提升从旧版本升级时的兼容性
+    - 降低 delta index 的内存使用量
+    - 使用更高效的 delta index update 算法
+
++ Tools
+
+    - BR
```
Comment (TomShawn) [inline releases/release-4.0.2.md:89]:
```suggestion
    - Backup & Restore (BR)
```

File: releases/release-4.0.2.md [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+    - 修复 TiDB 和对象存储服务之间 `BACKUP`/`RESTORE` 失败的问题 [#17844](https://github.com/pingcap/tidb/pull/17844)
+    - 修复权限检查失败时的错误信息 [#17724](https://github.com/pingcap/tidb/pull/17724)
+    - 修复 `DELETE`/`UPDATE` 语句的 feedback 被错误收集的问题 [#17843](https://github.com/pingcap/tidb/pull/17843)
+    - 禁止更改非 `AUTO_RANDOM` 表的 `AUTO_RANDOM_BASE` 值 [#17828](https://github.com/pingcap/tidb/pull/17828)
+    - 修复系统变量 `tidb_isolation_read_engines` 的值中没有 `tidb` 时某些系统表无法访问的问题 [#17719](https://github.com/pingcap/tidb/pull/17719)
+    - 修复 JSON 中大整数和浮点数比较的精度问题 [#17717](https://github.com/pingcap/tidb/pull/17717)
+    - 修复 `COUNT()` 函数的返回类型中 `DECIMAL` 不正确的问题 [#17704](https://github.com/pingcap/tidb/pull/17704)
+    - 修复 `HEX()` 函数的输入类型是二进制字符串时结果不正确的问题 [#17620](https://github.com/pingcap/tidb/pull/17620)
+    - 修复查询 `INFORMATION_SCHEMA.INSPECTION_SUMMARY` 表没有指定过滤条件时返回结果为空的问题 [#17697](https://github.com/pingcap/tidb/pull/17697)
+    - 修复 `ALTER USER` 语句使用哈希后的密码更新密码后，密码不符合预期的问题 [#17646](https://github.com/pingcap/tidb/pull/17646)
```
Comment (TomShawn) [inline releases/release-4.0.2.md:117]:
```suggestion
    - 修复 `ALTER USER` 语句使用哈希后的密码更新用户信息后，密码不符合预期的问题 [#17646](https://github.com/pingcap/tidb/pull/17646)
```

File: releases/release-4.0.2.md [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+    - 禁止在 sequence 缓存中出现负数 [#18103](https://github.com/pingcap/tidb/pull/18103)
+    - `CLUSTER_INFO` 表中不再显示 tombstone 状态的 TiKV 和 TiFlash 节点 [#17953](https://github.com/pingcap/tidb/pull/17953)
+    - 诊断规则 `current-load` 变更为 `node-check` [#17660](https://github.com/pingcap/tidb/pull/17660)
+
++ PD
+
+    - 持久化 `store-limit` 配置项，弃用 `store-balance-rate` 配置 [#2557](https://github.com/pingcap/pd/pull/2557)
+
+## 新功能
+
```
Comment (breeswish) [inline releases/release-4.0.2.md:26]:
```suggestion

## 新更改

- TiDB 及 TiDB Dashboard 默认收集使用情况信息，并将这些信息分享给 PingCAP 用于改善产品 [#18180](https://github.com/pingcap/tidb/pull/18180)。若要了解所收集的信息详情及如何禁用该行为，请参见[遥测](/telemetry.md)文档。

## 新功能

```

@TomShawn PTAL

File: releases/release-4.0.2.md [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+## Improvements
+
++ TiDB
+
+    - 降低当集群中 CM-Sketch 占用过多内存时，Golang 内存分配带来的查询延迟 [#17545](https://github.com/pingcap/tidb/pull/17545)
+    - 缩短 TiKV 故障恢复时集群 QPS 的恢复时间 [#17681](https://github.com/pingcap/tidb/pull/17681)
+    - 为 partition 表上的查询支持聚合函数下推到 TiKV 或者 TiFlash coprocessor [#17655](https://github.com/pingcap/tidb/pull/17655)
+    - 提升索引上等值条件的行数估算准确度 [#17611](https://github.com/pingcap/tidb/pull/17611)
+
++ TiKV
```
Comment (TomShawn) [inline releases/release-4.0.2.md:77]:
```suggestion
+ TiKV

```

File: releases/release-4.0.2.md (hunk 9 above)
Comment (TomShawn) [reply releases/release-4.0.2.md:26]: @uglyengineer PTAL

**Comment by TomShawn** [comment]
@uglyengineer Please also involve a technical review or PM review.

File: releases/release-4.0.2.md [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+    - 支持运行在 ARM 架构
+    - Coprocessor 支持 `JSON_LENGTH` 函数下推
+
++ TiCDC
+
+    - 支持 Capture 节点扩容时迁移部分子任务到新加节点 [#665](https://github.com/pingcap/ticdc/pull/665)
+    - Cli 中添加清理 TiCDC GC TTL 的功能 [#652](https://github.com/pingcap/ticdc/pull/652)
+    - 在 MQ sink 中支持输出 Canal 协议 [#649](https://github.com/pingcap/ticdc/pull/649)
+
+## Improvements
```
Comment (TomShawn) [inline releases/release-4.0.2.md:72]:
```suggestion
## 改进提升
```

File: releases/release-4.0.2.md [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+    - 修复 `tidb_replica_read` 设置成 `follower`，并且 Region 的 leader 和 follower/learner 之间出现网络分区后，TiDB 发送的 request 一直重试的问题 [#17443](https://github.com/pingcap/tidb/pull/17443)
+    - 修复某些情况下 TiDB 一直 ping PD 的 follower 的问题 [#17947](https://github.com/pingcap/tidb/pull/17947)
+    - 修复老版本的 range partition 表无法在 4.0 集群中加载的问题 [#17983](https://github.com/pingcap/tidb/pull/17983)
+    - 修复当多个 Region 的请求同时超时时整个 SQL 语句超时的问题 [#17585](https://github.com/pingcap/tidb/pull/17585)
+    - 修复解析日期类型的分隔符时和 MySQL 不兼容的问题 [#17501](https://github.com/pingcap/tidb/pull/17501)
+    - 修复少数情况下发给 TiKV 的请求错发给 TiFlash 的问题 [#18105](https://github.com/pingcap/tidb/pull/18105)
+
++ TiKV
+
+    - 修改 status server 的内存安全问题 [#8101](https://github.com/tikv/tikv/pull/8101)
```
Comment (Connor1996) [inline releases/release-4.0.2.md:139]:
```suggestion
    - 修复 status server 的内存安全问题 [#8101](https://github.com/tikv/tikv/pull/8101)
```

File: releases/release-4.0.2.md [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+    - 支持聚合函数 `APPROX_COUNT_DISTINCT()`，用于快速计算 `COUNT(DISTINCT)` 的近似值 [#18120](https://github.com/pingcap/tidb/pull/18120)
+    - TiFlash 支持了 collation，支持相应的函数下推 [#17705](https://github.com/pingcap/tidb/pull/17705)
+    - `INFORMATION_SCHEMA.INSPECTION_RESULT` 表新增 `STATUS_ADDRESS` 列，用于展示节点的 status 地址 [#17695](https://github.com/pingcap/tidb/pull/17695)
+    - `MYSQL.BIND_INFO` 表新增 `SOURCE` 列，用于展示 binding 的创建方式 [#17587](https://github.com/pingcap/tidb/pull/17587)
+    - `PERFORMANCE_SCHEMA.EVENTS_STATEMENTS_SUMMARY_BY_DIGEST` 表新增 `PLAN_IN_CACHE` 和 `PLAN_CACHE_HITS` 列，用于展示 plan cache 的使用情况 [#17493](https://github.com/pingcap/tidb/pull/17493)
+    - 新增配置项 `enable-collect-execution-info` 和会话级变量 `tidb_enable_collect_execution_info` 用于控制是否记录算子的执行信息并打印到慢查询日志中 [#18073](https://github.com/pingcap/tidb/pull/18073) [#18072](https://github.com/pingcap/tidb/pull/18072)
+    - 新增全局变量 `tidb_slow_log_masking`，用于控制是否脱敏慢查询日志中的查询 [#17694](https://github.com/pingcap/tidb/pull/17694)
+    - 增加对 TiKV 配置项 `storage.block-cache.capacity` 的诊断规则 [#17671](https://github.com/pingcap/tidb/pull/17671)
+    - 新增 SQL 语法 backup/restore 来进行备份恢复 [#15274](https://github.com/pingcap/tidb/pull/15274)
```
Comment (TomShawn) [inline releases/release-4.0.2.md:47]:
```suggestion
    - 新增 SQL 语法 `BACKUP`/`RESTORE` 来进行数据备份恢复 [#15274](https://github.com/pingcap/tidb/pull/15274)
```

**Comment by TomShawn** [comment]
LGTM

**Pull request merged**